packages/core/src/utils/utils.tsx

```diff
@@ -170,20 +187,29 @@ export const applyVariations = (componentName, variations = null) => (
  * @example getPaletteColor('primary.base')(props) => theme.palette.primary.base
  * @example getPaletteColor('primary', 'base')(props) => theme.palette.primary.base
  */
-export const getPaletteColor = (...args) => (props) => {
-  let color = args.length === 2 ? args[0] : props.color
-  let shade = args.length === 2 ? args[1] : args[0]
-
+export const getPaletteColor = (...args: string[]) => (props: {
+  theme: CreatedTheme
+  color: string
+  bg: string
+}): string => {
+  let color: string = args.length === 2 ? args[0] : props.color
+  let shade: string = args.length === 2 ? args[1] : args[0]
+
+  // eslint-disable-next-line @typescript-eslint/prefer-regexp-exec
   const colorShade = shade.match(/^([a-z]+)\.([a-z]+)$/i)
 
   if (colorShade) {
     color = colorShade[0]
     shade = colorShade[1]
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
```

Comment:
why are these necessary?

Reply:
The `themeGet` function from `styled-components` has return type `any`, and TS doesn't like that we're immediately invoking the result of those calls.